src/main.zig

```diff
@@ -765,85 +765,102 @@ fn buildOutputType(
+                        try cur_pkg.addAndAdopt(gpa, pkg_name.?, new_cur_pkg);
                         cur_pkg = new_cur_pkg;
                     } else if (mem.eql(u8, arg, "--pkg-end")) {
                         cur_pkg = cur_pkg.parent orelse
                             fatal("encountered --pkg-end with no matching --pkg-begin", .{});
                     } else if (mem.eql(u8, arg, "--main-pkg-path")) {
-                        if (i + 1 >= args.len) fatal("expected parameter after {s}", .{arg});
-                        i += 1;
-                        main_pkg_path = args[i];
+                        const next_arg = getNextArg(&process_args_iter, &resp_file_args_iter);
+                        if (next_arg == null) fatal("expected parameter after {s}", .{arg});
+                        main_pkg_path = next_arg.?;
                     } else if (mem.eql(u8, arg, "-cflags")) {
                         extra_cflags.shrinkRetainingCapacity(0);
                         while (true) {
-                            i += 1;
-                            if (i >= args.len) fatal("expected -- after -cflags", .{});
-                            if (mem.eql(u8, args[i], "--")) break;
-                            try extra_cflags.append(args[i]);
+                            const next_arg = getNextArg(&process_args_iter, &resp_file_args_iter);
+                            if (next_arg == null) fatal("expected -- after -cflags", .{});
+                            if (mem.eql(u8, next_arg.?, "--")) break;
+                            try extra_cflags.append(next_arg.?);
                         }
                     } else if (mem.eql(u8, arg, "--color")) {
-                        if (i + 1 >= args.len) {
-                            fatal("expected [auto|on|off] after --color", .{});
-                        }
-                        i += 1;
-                        const next_arg = args[i];
-                        color = std.meta.stringToEnum(Color, next_arg) orelse {
-                            fatal("expected [auto|on|off] after --color, found '{s}'", .{next_arg});
+                        const next_arg = getNextArg(&process_args_iter, &resp_file_args_iter);
+                        if (next_arg == null) fatal("expected [auto|on|off] after --color", .{});
+                        color = std.meta.stringToEnum(Color, next_arg.?) orelse {
+                            fatal("expected [auto|on|off] after --color, found '{s}'", .{next_arg.?});
                         };
                     } else if (mem.eql(u8, arg, "--subsystem")) {
-                        if (i + 1 >= args.len) fatal("expected parameter after {s}", .{arg});
-                        i += 1;
-                        if (mem.eql(u8, args[i], "console")) {
+                        const next_arg = getNextArg(&process_args_iter, &resp_file_args_iter);
+                        if (next_arg == null) fatal("expected parameter after {s}", .{arg});
```

Comment:
You could get rid of all the `.?`s below by using orelse here:
```suggestion
                        const next_arg = getNextArg(&process_args_iter, &resp_file_args_iter) orelse {
                            fatal("expected parameter after {s}", .{arg});
                        };
```
One could arguably use `orelse` in many other places in your patch as well, but this is the place where I see the biggest improvement.